src/ape/managers/config.py

```diff
@@ -19,6 +33,65 @@ class DeploymentConfig(ConfigItem):
     contract_type: str
 
 
+class DependencyConfig(ConfigItem):
+    name: str
+    _compilers: "CompilerManager" = None  # type: ignore
+
+    @abstractmethod
+    def extract_manifest(self) -> PackageManifest:
+        """
+        Download and extract the package manifest for this config entry.
+
+        Returns:
+            PackageManifest
+        """
+
+
+class IPFSDependency(DependencyConfig):
+    ipfs: str
+
+    def extract_manifest(self) -> PackageManifest:
+        response = requests.get(self.ipfs)
+        manifest_dict = response.json()
+        return PackageManifest(**manifest_dict)
+
+
+class GithubDependency(DependencyConfig):
+    github: str
+    version: str
+
+    def extract_manifest(self) -> PackageManifest:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_project_path = Path(temp_dir) / self.name
+            temp_project_path.mkdir(exist_ok=True, parents=True)
+            github_client.download_package(self.github, self.version, temp_project_path)
+
+            temp_contracts_path = temp_project_path / "contracts"
+            if not temp_contracts_path.exists():
+                raise ConfigError(
+                    "Dependency does not have a supported file structure. "
+                    "Expecting 'contracts/' path."
```

Comment:
a lot of dapptools/foundry projects will use `src/`

Reply:
Yeah I was wondering this.

This should be configurable on the `DependencyConfig`...
It can mirror our own `contracts_folder` perhaps